services/brig/src/Brig/Team/API.hs

```diff
@@ -230,69 +251,163 @@ createInvitation' tid mUid inviteeRole mbInviterUid fromEmail body = do
-  pending <- lift $ wrapClient $ DB.countInvitations tid
+  pending <- lift $ liftSem $ Store.countInvitations tid
   when (fromIntegral pending >= maxSize) $
     throwStd (errorToWai @'E.TooManyTeamInvitations)
 
   showInvitationUrl <- lift $ liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
 
-  lift $ do
-    iid <- maybe (liftIO DB.mkInvitationId) (pure . Id . toUUID) mUid
-    now <- liftIO =<< view currentTime
-    timeout <- setTeamInvitationTimeout <$> view settings
-    (newInv, code) <-
-      wrapClient $
-        DB.insertInvitation
-          showInvitationUrl
-          iid
-          tid
-          inviteeRole
-          now
-          mbInviterUid
-          email
-          body.inviteeName
-          timeout
-    (newInv, code) <$ sendInvitationMail email tid fromEmail code body.locale
-
-deleteInvitation :: (Member GalleyAPIAccess r) => UserId -> TeamId -> InvitationId -> (Handler r) ()
+  iid <- maybe (liftIO randomId) (pure . Id . toUUID) mUid
+  now <- liftIO =<< view currentTime
+  timeout <- setTeamInvitationTimeout <$> view settings
+  newInv <-
+    lift . liftSem $
+      Store.insertInvitation
+        iid
+        tid
+        inviteeRole
+        now
+        mbInviterUid
+        email
+        body.inviteeName
+        timeout
+  lift $ sendInvitationMail email tid fromEmail newInv.code body.locale
+  inv <- toInvitation showInvitationUrl newInv
+  pure (inv, newInv.code)
+
+deleteInvitation ::
+  (Member GalleyAPIAccess r, Member InvitationCodeStore r) =>
+  UserId ->
+  TeamId ->
+  InvitationId ->
+  (Handler r) ()
 deleteInvitation uid tid iid = do
   ensurePermissions uid tid [AddTeamMember]
-  lift $ wrapClient $ DB.deleteInvitation tid iid
+  lift . liftSem $ Store.deleteInvitation tid iid
 
-listInvitations :: (Member GalleyAPIAccess r) => UserId -> TeamId -> Maybe InvitationId -> Maybe (Range 1 500 Int32) -> (Handler r) Public.InvitationList
-listInvitations uid tid start mSize = do
+listInvitations ::
+  ( Member GalleyAPIAccess r,
+    Member TinyLog r,
+    Member InvitationCodeStore r
+  ) =>
+  UserId ->
+  TeamId ->
+  Maybe InvitationId ->
+  Maybe (Range 1 500 Int32) ->
+  (Handler r) Public.InvitationList
+listInvitations uid tid startingId mSize = do
   ensurePermissions uid tid [AddTeamMember]
   showInvitationUrl <- lift $ liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
-  rs <- lift $ wrapClient $ DB.lookupInvitations showInvitationUrl tid start (fromMaybe (unsafeRange 100) mSize)
-  pure $! Public.InvitationList (DB.resultList rs) (DB.resultHasMore rs)
+  let toInvitations is = mapM (toInvitation showInvitationUrl) is
+  lift (liftSem $ Store.lookupInvitationsPaginated mSize tid startingId) >>= \case
+    PaginatedResultHasMore storedInvs -> do
+      invs <- toInvitations storedInvs
+      pure $ InvitationList invs True
+    PaginatedResult storedInvs -> do
+      invs <- toInvitations storedInvs
+      pure $ InvitationList invs False
```

Comment:
Is this the pagination type newly introduced in this PR? Why not `Cassandra.Page`? It's already there, and it allows for nicer code here because it lets you get to the page and the `hasMore` flag in any order.

Reply:
Two reasons: stores are abstract interfaces and should ideally not depend on Cassandra, I know it does in practice but we should at least  try and make it as separated as possible and at least not have it depend on the Cassandra packages, afaiu. 

Secondly, @elland doesnt like Bool, which is fair 😅